File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        const map = new Map()
        for(let [i,num] of numbers.entries()){
            const required = target-num
            if(map.has(required)){
                return [map.get(required), i+1]
            }
            map.set(num,i+1)
        }
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        freq = {}
        for i in range(len(nums)):
            req = target-nums[i]
            if req in freq:
                return [freq[req],i]
            freq[nums[i]] = i
        return [-1,-1]
        